src/metabase/pulse.clj

```diff
@@ -300,18 +312,25 @@
                                            :user-id (:creator_id pulse)
                                            :object  {:recipients (map :recipients (:channels pulse))
                                                      :filters    (:parameters pulse)}})
-        (u/prog1 (doseq [channel channels]
+        (u/prog1 (doseq [pulse-channel channels]
                    (try
-                     (let [channel-type (if (= :email (keyword (:channel_type channel)))
-                                            :channel/email
-                                            :channel/slack)
-                           messages     (channel/render-notification channel-type
-                                                                     (get-notification-info pulse parts channel)
-                                                                     (channels-to-channel-recipients channel))]
-                       (doseq [message messages]
-                         (send-retrying! {:type channel-type} message)))
+                     (let [channel  (pc->channel pulse-channel)
+                           messages (channel/render-notification (:type channel)
+                                                                 (get-notification-info pulse parts pulse-channel)
+                                                                 (channel-recipients pulse-channel))]
+                       (log/debugf "Rendered messages for %s %d to channel %s. Got %d"
+                                   (alert-or-pulse pulse)
+                                   (:id pulse)
+                                   (:type channel)
+                                   (count messages))
```

Comment:
"What did you get X of"? I think reordering this a bit makes it somewhat clearer:

```suggestion
                       (log/debugf "Rendered %d messages for %s %d to channel %s."
                                   (count messages)
                                   (alert-or-pulse pulse)
                                   (:id pulse)
                                   (:type channel))
```

Reply:
Thanks! 3ba4207d92